### PR TITLE
Charlesmchen/custom notification sounds3

### PR DESCRIPTION
**Pull request by charlesmchen**
This PR is now ready for review.

PTAL @michaelkirk

### CODE REVIEW

**Comment by charlesmchen** [comment]
This PR is now ready for review.

PTAL @michaelkirk

**Comment by charlesmchen** [comment]
See the commit messages for a summary of the changes.

File: Signal/src/ViewControllers/OWSSoundSettingsViewController.h [hunk 1]
```diff
@@ -6,10 +6,7 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-typedef NS_ENUM(NSUInteger, OWSSoundType) {
-    OWSSoundType_Notification = 0,
-    OWSSoundType_Ringtone,
-};
+typedef NS_ENUM(NSUInteger, OWSSoundType) { OWSSoundType_Notification = 0 };
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/OWSSoundSettingsViewController.h]: Does this enum make sense anymore?

We now have switches that only have one case - getting rid of it seems like a nice little cleanup.

File: Signal/src/call/Speakerbox/CallKitCallUIAdaptee.swift [hunk 2]
```diff
@@ -124,12 +126,6 @@ final class CallKitCallUIAdaptee: NSObject, CallUIAdaptee, CXProviderDelegate {
 
         disableUnsupportedFeatures(callUpdate: update)
 
-        // Update the provider configuration to reflect the caller's ringtone.
-        let sound = OWSSounds.ringtoneSound(for: call.thread)
-        let providerConfiguration = provider.configuration
-        providerConfiguration.ringtoneSound = OWSSounds.filename(for: sound)
```
Comment (michaelkirk) [inline Signal/src/call/Speakerbox/CallKitCallUIAdaptee.swift]: conflict heads up - this code has been removed in master already (https://github.com/signalapp/Signal-iOS/commit/c66fb70b2f2cf477adffa045e2a742bc1f46efcf)

File: SignalMessaging/environment/OWSSounds.m [hunk 3]
```diff
@@ -148,62 +113,12 @@ + (NSString *)displayNameForSound:(OWSSound)sound
             return @"Pulse";
         case OWSSound_Synth:
             return @"Synth";
+        case OWSSound_ClassicNotification:
+            return @"Classic";
```
Comment (michaelkirk) [inline SignalMessaging/environment/OWSSounds.m]: alternative name proposal: "Kickin the Bucket"

File: SignalMessaging/environment/migrations/OWSDatabaseMigration.m [hunk 4]
```diff
@@ -50,10 +50,21 @@ - (void)runUpWithCompletion:(OWSDatabaseMigrationCompletion)completion
...
-            [self runUpWithTransaction:transaction];
-        }
+    OWSDatabaseConnection *dbConnection = (OWSDatabaseConnection *)self.storageManager.newDatabaseConnection;
+    // These migrations won't be run until storage registrations are enqueued,
+    // but this transaction might begin before all registrations are marked as
+    // complete, so disable this checking.
+    //
+    // TODO: Once we move "app readiness" into AppSetup, we should explicitly
+    // not start these migrations until storage is ready.  We can then remove
+    // this statement which disables checking.
```
Comment (michaelkirk) [inline SignalMessaging/environment/migrations/OWSDatabaseMigration.m]: :+1: good comment

File: SignalMessaging/utils/OWSAudioPlayer.m [hunk 5]
```diff
@@ -92,6 +117,9 @@ - (void)play
             return;
         }
         self.audioPlayer.delegate = self;
+        if (self.isLooping) {
+            self.audioPlayer.numberOfLoops = -1;
```
Comment (michaelkirk) [inline SignalMessaging/utils/OWSAudioPlayer.m]: 👍

File: Signal/Images.xcassets/table_ic_notification_sound.imageset/Contents.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "filename" : "table_ic_notification_sound@1x.png",
```
Comment (michaelkirk) [inline Signal/Images.xcassets/table_ic_notification_sound.imageset/Contents.json]: IMO this looks too related to the "Mute" icon. What do you think about using music note?

File: Signal/src/ViewControllers/OWSSoundSettingsViewController.h (hunk 1 above)
Comment (charlesmchen) [reply Signal/src/ViewControllers/OWSSoundSettingsViewController.h]: Done.  I was thinking we might use it again...

File: Signal/src/call/Speakerbox/CallKitCallUIAdaptee.swift (hunk 2 above)
Comment (charlesmchen) [reply Signal/src/call/Speakerbox/CallKitCallUIAdaptee.swift]: k

File: SignalMessaging/environment/OWSSounds.m (hunk 3 above)
Comment (charlesmchen) [reply SignalMessaging/environment/OWSSounds.m]: "Your screen's cracked and your speaker's cracked too."

File: Signal/Images.xcassets/table_ic_notification_sound.imageset/Contents.json (hunk 6 above)
Comment (charlesmchen) [reply Signal/Images.xcassets/table_ic_notification_sound.imageset/Contents.json]: A capital idea.  What think you of this icon?

simulator screen shot - iphone x - 2018-02-26 at 15 06 33

**Pull request merged**